bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -48,7 +48,7 @@ struct RequestState {
 };
 
 // Pre-allocated (clientId * dataSize) buffers
-typedef std::vector<concordUtils::Sliver> PreProcessResultBuffers;
+typedef std::unordered_map<uint16_t, std::pair<std::atomic_bool, concordUtils::Sliver>> PreProcessResultBuffers;
```

Comment:
Didn't we intend to return it to be a vector?

Reply:
Done

Reply:
Agreed to use deque since we cant use vector with atomic bool.